omegaconf/grammar_parser.py

```diff
@@ -115,10 +116,10 @@ def parse(
         # from antlr4 import PredictionMode
         # parser._interp.predictionMode = PredictionMode.SLL
 
-        _grammar_cache = lexer, tokens, parser
+        _grammar_cache.data = lexer, tokens, parser
```

Comment:
1. Can you rename tokens to tokenizer or token_stream?
2. I find it a bit odd that we are caching the input stream (via the lexer).
Can you explain the caching strategy?

Reply:
1. Done in a5c66fc
2. At high-level the strategy is "let's try to cache the lexer and parser so that we don't need to re-create them every time we parse a new string". Caching the input stream is just a side effect of caching the lexer, we don't really need it since we override it when we re-use the cached lexer. I guess we could set `lexer.inputStream = None` after parsing to avoid keeping a reference to the input string -- it'd be very surprising for it to ever matter, but I can easily add it if you think it's better.

Reply:
This is more about the semantics than about an actual runtime concern.
A comment explaining that we are replacing the input stream when we reuse the cache would be sufficient.

Reply:
Got it -- done in 2b4aba8